### PR TITLE
Feat: fix load add thiserror

### PR DESCRIPTION
**Pull request by LeanSerra**
**This PR**
- Fix the `load_memory` function it was ignoring the origin address.
- Now the `VMError` uses the thiserror crate
- Change the name from of the enum from `opcodes` to `opcode`
- Move the flags enum to it's own file

### CODE REVIEW

**Comment by lima-limon-inc** [comment]
On it

File: src/lc3_vm/flags.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#[allow(clippy::upper_case_acronyms)]
+enum ConditionFlags {
+    POS = 0,
+    ZRO = 2,
+    NEG = 4,
```
Comment (lima-limon-inc) [inline src/lc3_vm/flags.rs:5]: Is the value here necessary? You translate the values to u16 in the `From<ConditionFlags>` function; and they are the same values. 

Would there be a problem if these values get out-of-sync?

**Review by lima-limon-inc — APPROVED** [review]

File: src/lc3_vm/flags.rs (hunk 1 above)
Comment (dsocolobsky) [inline src/lc3_vm/flags.rs:5]: This might not be necessary but you can change it later

**Review by dsocolobsky — APPROVED** [review]

**Review by damiramirez — APPROVED** [review]
LGTM

**Pull request merged**